templates/todo/projects/csharp-cosmos-sql-func/.vscode/launch.json

```diff
@@ -0,0 +1,33 @@
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "Debug Web",
+            "request": "launch",
+            "type": "pwa-msedge",
+            "webRoot": "${workspaceFolder}/src/web/src",
+            "url": "http://localhost:3000",
+            "sourceMapPathOverrides": {
+                "webpack:///src/*": "${webRoot}/*"
+            },
+        },
+        {
+            "name": "Debug API",
+            "type": "coreclr",
+            "request": "attach",
+            "processId": "${command:azureFunctions.pickProcess}"
+        },
+        {
+            "name": ".NET Core Attach",
+            "type": "coreclr",
+            "request": "attach"
+        }
+    ],
+    "inputs": [
+        {
+            "id": "dotEnvFilePath",
```

Comment:
This does not seem to be used by anything. Delete?

Reply:
I've realized that the Debug API launch is not loading environment variables, so all debugging will throw an error. I can't figure out how to get this to work. At the moment, "Debug API" will attach itself to the process that comes out of command:azureFunctions.pickProcess. This command executes the "API Functions run" task. However, that task will only have environment variables when it's run by the Start API task. Any suggestions here?

Reply:
You should be able to use the `getDotEnvFilePath` command like other templates do; just refer to it from the debug configuration. For example https://github.com/Azure-Samples/todo-nodejs-mongo/blob/main/.vscode/launch.json#L31

Reply:
If you are using attach, you should ensure that env vars are populated when the process is launched, which is a separate step:

1. Start the API server using "Start API" task
2. Then attach the debugger using "Debug API" launch config

Using "Start API" task should do the trick--I see you are using the dotenv task type, which should populate the process env. Is that not working?

Reply:
I don't have a way of getting "Debug API" in launch.json to run "Start API" to populate the dotenv variables. As we discussed the preLaunchTask doesn't work in this case. If you find a better way I'd be happy to implement it. 

The alternative I've found and is in this commit is set the "Debug API" "processId" to blank. This will cause VS Code to ask the user to select the process to attach the debugger to. We would need to instruct the user to manually run the "Start API" task, then run the debugger and select the dotnet process. I can provide these instructions too.

Reply:
@karolz-ms Thank you for the guidance! I've figured it out and the change is in the latest commit.